vulnerabilities/importers/yubico.py

```diff
@@ -0,0 +1,193 @@
+#
+
+import logging
+import re
+from datetime import datetime
+from datetime import timezone
+
+import requests
+from bs4 import BeautifulSoup
+from packageurl import PackageURL
+from vulnerabilities.utils import get_item
+from vulnerabilities.utils import fetch_response
+from urllib.parse import urljoin
+
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
+from vulnerabilities.importer import Importer
+from univers.versions import SemverVersion
+from vulnerabilities.importer import Reference
+from vulnerabilities.importer import VulnerabilitySeverity
+from vulnerabilities.severity_systems import SCORING_SYSTEMS
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__) 
+
+class YubicoImporter(Importer):
+
+    root_url = "https://www.yubico.com/support/security-advisories/"
+    spdx_license_expression = "NOASSERTION"
+    importer_name = "Yubico Security Bulletin Importer"
+
+    def advisory_data(self):
+        urls = fetch_links(self.root_url)
+        for url in urls:
+            yield self.to_advisory(url)
+
+    def to_advisory(self,url):
+        output_generated = get_adv_data(url)
+
+        severity = VulnerabilitySeverity(
+            system=SCORING_SYSTEMS["cvssv3.1"],
+            value=get_item(output_generated,"score")
+        )
+
+        reference = Reference(
+            reference_id=get_item(output_generated,"cve"),
+            severities=severity,
+            url=url
+        )
+
+        affected_packages = []
+        try:
+            for affected_package in get_item(output_generated,"affected_packages"):
+                print(affected_package)
+                fixed_version = SemverVersion(get_item(affected_package,"version"))
+                affected_packages.append(AffectedPackage(
+                    package=PackageURL(
+                        type="generic",
+                        name=get_item(affected_package,"package")
+                    ),
+                    fixed_version=fixed_version
+                ))
+        except:
+            pass
+
+        try:
+            date_published = datetime.strptime(
+                get_item(output_generated,"published_date") or "", "%Y-%m-%d"
+            ).replace(tzinfo=timezone.utc)
+        except:
+            date_published =''
+
+        return AdvisoryData(
+            aliases=get_item(output_generated,'cve'),
+            summary=get_item(output_generated,'summary'),
+            affected_packages=affected_packages,
+            references=reference,
+            date_published=date_published,
+            url=url
+        )
+
+
+
+#to fetch all the links in advisory page 
+'''
+link structure : https://www.yubico.com/support/security-advisories/ysa-2024-03/
+href : /support/issue-rating-system/security-advisories/ysa-2017-01/
+       https://www.yubico.com/support/security-advisories/ysa-2025-01/
+'''
+root_url = "https://www.yubico.com/support/security-advisories/"
```

Comment:
Do not introduce new conventions/styles, make sure your code conforms to the existing code style and formatting.